sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java

```diff
@@ -594,7 +612,17 @@ private Bound(String name, String filenamePrefix, String filenameSuffix, Coder<T
        * <p>Does not modify this object.
        */
       public Bound<T> withoutValidation() {
-        return new Bound<>(name, filenamePrefix, filenameSuffix, coder, numShards,
+        return new Bound<>(name, filenamePrefix, filenameSuffix, header, footer, coder, numShards,
+            shardTemplate, false);
+      }
+
+      public Bound<T> withHeader(String header) {
+        return new Bound<>(name, filenamePrefix, filenameSuffix, header, footer, coder, numShards,
+            shardTemplate, false);
+      }
+
+      public Bound<T> withFooter(String footer) {
```

Comment:
Footer should be of T type to support TextIO withCoder usage which aren't strings.